apps/wpcom-block-editor/src/wpcom/features/utils.js

```diff
@@ -171,3 +171,20 @@ export const buildGlobalStylesContentEvents = debounce( ( updated, original, eve
 		tracksRecordEvent( eventName, buildGlobalStylesEventProps( keyMap, value ) );
 	} );
 }, 100 );
+
+export const getFlattenedBlockNames = ( block ) => {
+	const blockNames = [];
+
+	const queue = Array.isArray( block ) ? [ ...block ] : [ block ];
+	while ( queue.length > 0 ) {
+		const currentBlock = queue.pop();
+
+		blockNames.push( currentBlock.name );
+
+		for ( const innerBlock of currentBlock.innerBlocks ) {
+			queue.unshift( innerBlock );
```

Comment:
It seems like we are adding inner blocks in an odd order here.  I think either we should make the `queue.pop()` a `queue.shift()` to move in order from the front of the block list, or alternatively 
 `blockNames.push` to `blockNames.unshift` and `queue.unshift` to `queue.push` so its moving backwards and appending on the front of the list.

For example, I have converted/detached a paragraph block followed by a buttons block (with a button inner block).  The resulting order in this current implementation is `core/buttons, core/paragraph, core/button`, where one might expect `core/paragraph, core/buttons, core/button`.